terps/geas/GeasRunner.hh

```diff
@@ -109,49 +109,49 @@ protected:
   virtual GeasResult set_style (const GeasFontStyle &) { return r_not_supported; }
 
 public:
-  virtual std::string absolute_name (std::string rel_name, std::string parent) const = 0;
-  virtual std::string get_file (std::string filename) const = 0;
-  virtual void debug_print (std::string s) { std::cerr << s << std::endl; }
-  virtual GeasResult wait_keypress (std::string) { return r_not_supported; }
+  virtual std::string absolute_name (const std::string &rel_name, const std::string &parent) const = 0;
+  virtual std::string get_file (const std::string &filename) const = 0;
+  virtual void debug_print (const std::string &s) { std::cerr << s << std::endl; }
+  virtual GeasResult wait_keypress (const std::string &) { return r_not_supported; }
   virtual GeasResult pause (int msec) { return r_not_supported; }
   virtual GeasResult clear_screen () { return r_not_supported; }
 
   //virtual GeasResult set_foreground (string) { return r_not_supported; }
   //virtual GeasResult set_background (string) { return r_not_supported; }
-  virtual void set_foreground (std::string) = 0;
-  virtual void set_background (std::string) = 0;
-  void set_default_font_size (std::string s);
-  void set_default_font (std::string s);
+  virtual void set_foreground (const std::string &) = 0;
+  virtual void set_background (const std::string &) = 0;
+  void set_default_font_size (const std::string &s);
+  void set_default_font (const std::string &s);
 
   /* Unsure what arguments this will take.
    * May also add animated, persistent, close image
    */
-  virtual GeasResult show_image (std::string filename, std::string resolution,
-				 std::string caption, ...)
+  virtual GeasResult show_image (const std::string &filename, const std::string &resolution,
+				 const std::string &caption, ...)
   { return r_not_supported; }
 
   /* Again, unsure what arguments to give
    * May add sound type
    * If sync is true, do not return until file ends
    * If filename is "", stop playing sounds.
    */
-  virtual GeasResult play_sound (std::string filename, bool looped, bool sync)
+  virtual GeasResult play_sound (const std::string &filename, bool looped, bool sync)
   { return r_not_supported; }
-
+  
   /* Asks the user to type a free format string
    */
   virtual std::string get_string () = 0;
-
+  
```

Comment:
A couple lines here had extraneous whitespace added; a very minor point, but I think worth retaining the original "just a newline".